trieste/models/gpflow/utils.py

```diff
@@ -76,6 +68,18 @@ def randomize_hyperparameters(object: gpflow.Module) -> None:
                 sample = param.prior.sample(tf.shape(param))
             else:
                 sample = param.prior.sample()
+            if param.prior_on is gpflow.base.PriorOn.UNCONSTRAINED:
+                param.unconstrained_variable.assign(sample)
+            else:
+                param.assign(sample)
```

Comment:
There's a risk here that we hit a commonly encountered error where we can't assign a value incompatible with the transform. This could be eg if we have a Gaussian prior on a sigmoid transform-ed parameter. But that is the user's responsibility to take so I think it's fine here.

Reply:
we could add a note in the docstring about it

Reply:
Added